packages/teleterm/src/ui/services/usage/askAboutUserJobRoleIfNeeded.ts

```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright 2023 Gravitational, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { UsageService } from 'teleterm/ui/services/usage';
+import { ModalsService } from 'teleterm/ui/services/modals';
+import { StatePersistenceService } from 'teleterm/ui/services/statePersistence';
+import { ConfigService } from 'teleterm/services/config';
+
+export async function askAboutUserJobRoleIfNeeded(
+  statePersistenceService: StatePersistenceService,
+  configService: ConfigService,
+  modalsService: ModalsService,
+  usageService: UsageService
+): Promise<void> {
+  const { askedForUserJobRole } =
+    statePersistenceService.getUsageReportingState();
+
+  if (
+    askedForUserJobRole ||
+    !configService.get('usageReporting.enabled').value
+  ) {
+    return;
+  }
+
+  await showUserJobRoleDialog(modalsService, usageService);
+  statePersistenceService.saveUsageReportingState({
+    askedForUserJobRole: true,
+  });
+}
+
+function showUserJobRoleDialog(
+  modalsService: ModalsService,
+  usageService: UsageService
+): Promise<void> {
+  return new Promise(resolve => {
+    modalsService.openRegularDialog({
+      kind: 'user-job-role',
+      onSend(jobRole) {
+        usageService.captureUserJobRoleUpdate(jobRole);
+        resolve();
```

Comment:
Another small nit: It's kind of annoying that we have to write those custom promise wrappers for each modal. In an ideal world, we'd have an uniform way of doing that (though idk how well it'd handle edge cases such as `UsageData` with three different callbacks).

My point being, when we write a custom promise wrapper like this, it might be worthwhile to keep the logic in it to minimum, resolve with a value (`jobRole` in this case) and then implement the rest of the logic outside of the wrapper. Meaning we'd await `jobRole` and then call `usageService.captureUserJobRoleUpdate`.

This way the whole logic could be laid out in `askAboutUserJobRoleIfNeeded` and you could just read that function top to bottom to understand what's going on, without having to jump into `showUserJobRoleDialog` to see that there's a side effect hapenning in `onSend`.

Reply:
Yeah, it is quite inconvenient to have to write that _promisify_ function, but nothing better comes to my mind :(
To have a generic wrapper we should first have a standard set of callbacks passed to modals.

I like the suggestion about moving the logic to the parent function, done.